### DISCUSSION TITLE
Release BINARY distribution as Homebrew tap

### DISCUSSION THREAD

**Discussion by helpermethod** [jreleaser/jreleaser#247] [Q&A] [answered]
Hi!

I'm trying to release my BINARY distribution (consisting of 3 native-images packaged as ZIP).

Two questions:

1. Can I perform a Homebrew release using a personal access token with scope `repo`?
2. Do I need to tell JReleaser which artifact to release as a Homebrew tap?

I've added the following two lines to my distribution 

```
brew:
      active: ALWAYS
```

And added a secret named `JRELEASER_HOMEBREW_GITHUB_TOKEN` containing
the personal access token. 

This is the output I'm seeing:

```
[INFO] Publishing distributions 

[INFO] - Publishing connect-offset-reset distribution 

[INFO] [brew] publishing connect-offset-reset distribution 

[INFO] [brew] setting up repository helpermethod/homebrew-tap 

[INFO] Writing output properties to out/jreleaser/output.properties 

Error: JReleaser failed after 6.812 s 

Unexpected error 

61Unexpected error updating helpermethod/homebrew-tap 

62Error: Process completed with exit code 1.
```

**Comment by aalmiray** [comment, marked as answer]
I suppose you may use either BINARY or NATIVE_IMAGE as distribution type as both support the same packagers at this moment.

1. Yes. Homebrew taps will reuse the same `JRELEASER_GITHUB_TOKEN` by default if no matching `JRELEASER_HOMREBREW_GITHUB_TOKEN` is configured.
2. No need, brew supports either universal (no platform) or `osx` artifacts at the moment. Once we get https://github.com/jreleaser/jreleaser/issues/215 then `linux` artifacts will be supported automatically, similarly as shown by https://github.com/kordamp/homebrew-tap/blob/master/Formula/gum.rb

Could you post the stacktrace found at `out/jreleaser/trace.log`?

**Reply by helpermethod** [reply]
Sorry for the late reply.

First I forgot to set JRELEASER_GITHUB_TOKEN, then I've used an environment secret instead of a repository secret.

Now everything is working fine 🤘!
